### PR TITLE
DPC-1479: Replicate in admin portal - login page

### PR DESCRIPTION
**Pull request by Sun-Mountain**
### Fixes [DPC-1479](https://jira.cms.gov/browse/DPC-1479)

Admin users should be able to sign into their Github account to enter the admin portal.

### Proposed Changes

- Replicate current admin login page
- Create existing Github point and port number (4002)

### Change Details

- Add rspec tests
- Add rubocop
- Add devise/omniauth gems
- Add simplecov and climate_control gems
- Tests for creating and signing in as admin
- Admin table in database
- READMEs for `dpc-impl` and `dpc-adminv2`

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Screen Shot 2021-06-22 at 1 54 19 PM

Screen Shot 2021-06-22 at 1 54 28 PM

### CODE REVIEW

File: dpc-adminv2/app/views/auth/unlocks/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend unlock instructions</h2>
```
Comment (dhgreene) [inline dpc-adminv2/app/views/auth/unlocks/new.html.erb:1]: Is it possible to have an OAuth-based user locked?

File: dpc-adminv2/app/views/shared/_navbar.html.erb [hunk 2]
```diff
@@ -30,5 +30,15 @@ aria-controls="topnav-wrap" aria-label="Close menu">
       <li class="topnav__item">
         <%= link_to "Frequently Asked Questions", ENV.fetch('STATIC_SITE_URL') + 'faq', class: "topnav__link" %>
       </li>
+
+      <% unless (controller_name == "sessions" || controller_name == "registrations") %>
```
Comment (dhgreene) [inline dpc-adminv2/app/views/shared/_navbar.html.erb:34]: Will we eventually have a `registrations` controller for admin users?

File: dpc-adminv2/app/views/auth/shared/_links.html.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  <%= link_to "Forgot your password?", new_password_path(resource_name) %><br />
+<% end %>
+
+<%- if devise_mapping.confirmable? && controller_name != 'confirmations' %>
+  <%= link_to "Didn't receive confirmation instructions?", new_confirmation_path(resource_name) %><br />
+<% end %>
+
+<%- if devise_mapping.lockable? && resource_class.unlock_strategy_enabled?(:email) && controller_name != 'unlocks' %>
+  <%= link_to "Didn't receive unlock instructions?", new_unlock_path(resource_name) %><br />
+<% end %>
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-adminv2/app/views/auth/shared/_links.html.erb:21]: Not currently, May need them in the future depending how we want to sign in in the future.

**Review by dhgreene — APPROVED** [review]
🛂

**Pull request merged**